app/models/work_version.rb

```diff
@@ -265,6 +265,11 @@ def update_doi?
     super && latest_published_version?
   end
 
+  def null_doi?
+    !latest_published_version? || (latest_published_version? &&
+        work.fields_with_dois.map { |field| work.send(field).present? }.include?(true))
+  end
+
```

Comment:
There are two changes we can do here.

First is an easy one, `work.fields_with_dois.map { |field| work.send(field).present? }.include?(true)` basically just re-implements `#all_dois` so we can change this to `work.all_dois.any?`

The second thing we can do is delete this method, but use the logic that you wrote to override `#all_dois` for `WorkVersion` like so:

```ruby
class WorkVersion < ApplicationRecord
  # ...

  def all_dois
    return [] if !latest_published_version? || 
                   (latest_published_version? && work.all_dois.any?)

    super
  end

 #...
end

```